3-With-Intune MAM/IntuneClient/active-directory-xamarin-intune/PCAWrapper.cs

```diff
@@ -0,0 +1,119 @@
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Xamarin.Forms;
+
+namespace activedirectoryxamarinintune
+{
+    /// <summary>
+    /// This is a wrapper for PCA. It is singleton and can be utilized by both application and the MAM callback
+    /// </summary>
+    public class PCAWrapper
+    {
+
+        /// <summary>
+        /// This is the singleton used by consumers
+        /// </summary>
+        static public PCAWrapper Instance { get; }
+
+        internal IPublicClientApplication PCA { get; }
+
+        // This event coordinates between different callbacks
+        public static ManualResetEvent MAMRegsiteredEvent { get; } = new ManualResetEvent(false);
+
+        /// <summary>
+        /// The authority for the MSAL PublicClientApplication. Sign in will use this URL.
+        /// </summary>
+        private const string _authority = "https://login.microsoftonline.com/organizations";
+
+        // ClientID of the application in (msidlab4.com)
+        private static string _clientID = "94996db4-9d57-422d-a707-84a1328a3cb8"; // TODO - Replace with your client Id. And also replace in the AndroidManifest.xml
+
+        // TenantID of the organization (msidlab4.com)
+        private static string _tenantID = "f645ad92-e38d-4d1a-b510-d1b09a74a8ca"; // TODO - Replace with your TenantID. And also replace in the AndroidManifest.xml
+
+        static string[] clientCapabilities = { "ProtApp" }; // It is must to have these capabilities
+
+        // private constructor for singleton
+        private PCAWrapper()
+        {
+            // Create PCA once. Make sure that all the config parameters below are passed
+            // ClientCapabilities - must have ProtApp
+            PCA = PublicClientApplicationBuilder
+                                        .Create(_clientID)
+                                        .WithAuthority(_authority)
+                                        .WithBroker()
+                                        .WithClientCapabilities(clientCapabilities)
+                                        .WithTenantId(_tenantID)
+                                        .WithRedirectUri(PlatformConfigImpl.Instance.RedirectUri)
+                                        .WithIosKeychainSecurityGroup("com.microsoft.adalcache")
+                                        .Build();
+        }
+
+        /// <summary>
+        /// Static constructor to instantiate PCA
+        /// </summary>
+        static PCAWrapper()
+        {
+            Instance = new PCAWrapper();
+        }
+
+        /// <summary>
+        /// Perform the intractive acquistion of the token for the given scope
+        /// </summary>
+        /// <param name="scopes">desired scopes</param>
+        /// <param name="parentWindow">Parent window</param>
+        /// <returns></returns>
+        internal async Task<AuthenticationResult> DoInteractiveAsync(string[] scopes)
```

Comment:
Maybe we should call these like the MSAL methods, i.e. `AcquireTokenInteractive` ? To reduce confusion.